### ISSUE TITLE
Failed to build RiderLink plugin 2020.3.113

### ISSUE THREAD

**Issue by RVillani** [JetBrains/UnrealLink#62]
I've updated Rider today and the plugin asked to be synchronized. But it keeps failing.
I don't understand the logs, and it's something coming from CS files. I hope the team can shed some light upon this matter.

[All logs are here](https://github.com/JetBrains/UnrealLink/files/5604263/log.zip)

I deleted the logs before trying once more to have a clean slate on the files. I hope it helps.

**Comment by DecoyRS** [comment]
Hello @RVillani !
Unfortunatelly, your logs are mostly empty folders...
Let's try one thing though
1. Backup old version of RiderLink
2. Find {R4UE_InstallFolder}\plugins\UnrealLink\EditorPlugin\RiderLink.zip
3. Unzip it in place of old version of RiderLink
4. Try to build your project as usual

**Comment by RVillani** [comment]
Thank you, @DecoyRS!

I have to install the plugin in engine, to not affect others working on the project with VS.

Fortunately, I'm working on the source version from Git, so I tried replacing the files, executing GenerateProjectFiles.bat, and building the engine.

That made UnrealBuildTool throw this error, before doing any work:
```
2>UnrealBuildTool: Error  : Unable to merge actions producing instantiations.cpp.obj: prerequisites are different
2>UnrealBuildTool: Error  : Unable to merge actions producing instantiations.cpp.obj: prerequisites are different
2>UnrealBuildTool: Error  : Action graph is invalid; unable to continue. See log for additional details.
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.MakeFile.Targets(46,5): Error MSB3073 : The command "C:\WINDOWS\System32\chcp.com 65001 >NUL 
 C:\UE4_Git\Engine\Build\BatchFiles\Build.bat -Target="BravoEditor Win64 Development -Project=\"D:\Offworld\bravo\Bravo.uproject\"" -Target="ShaderCompileWorker Win64 Development -Quiet" -WaitMutex -FromMsBuild" exited with code 6.
```
After investigation, I found that means that UE4 doesn't appreciate having two cpp files with same name on the same module.

One of the instantiations.cpp (and its header) were mostly empty (the ones inside Public/Model/RdEditorProtocol). So I deleted them and the engine is now building.

It's currently on 350/4854 actions and it didn't get to the RiderLink objects yet, so I'll post here once I find out if it worked.

**Comment by RVillani** [comment]
The engine hasn't completed building yet, but both RiderLink.cpp and instantiations.cpp files are already built without errors nor warnings. Therefore, I think it's safe to assume deleting instantiations.cpp and .h from RdEditorProtocol worked.

**Comment by RVillani** [comment]
@DecoyRS, for other people with problems installing the plugin, would you say deleting those `instantiations` files from the original zip on the R4UE folder would suffice? Does Rider copy the files directly from there?

**Comment by DecoyRS** [comment]
yes, Rider simply unzip's bundled RiderLink.zip into temp folder and tries to build plugin there.
I plan to build a new version of plugin with the hotfix, it'll be available soon.

**Comment by DecoyRS** [comment]
Fixed in 2020.3.116
